platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/logging/LogLine.java

```diff
@@ -0,0 +1,468 @@
+import static com.swirlds.cli.logging.HtmlGenerator.THREAD_NAME_COLUMN_LABEL;
+import static com.swirlds.cli.logging.HtmlGenerator.TIMESTAMP_COLUMN_LABEL;
+import static com.swirlds.cli.logging.HtmlGenerator.WHITELIST_LABEL;
+import static com.swirlds.cli.logging.LogProcessingUtils.escapeString;
+import static com.swirlds.cli.logging.LogProcessingUtils.getLogLevelColor;
+import static com.swirlds.cli.logging.LogProcessingUtils.parseTimestamp;
+import static com.swirlds.common.units.TimeUnit.UNIT_MILLISECONDS;
+import static com.swirlds.common.units.TimeUnit.UNIT_SECONDS;
+
+import com.swirlds.common.formatting.TextEffect;
+import com.swirlds.common.formatting.UnitFormat;
+import com.swirlds.common.formatting.UnitFormatter;
+import com.swirlds.common.system.NodeId;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Represents a single line of log output
+ * <p>
+ * 2023-05-18 09:03:03.618     80         INFO    PLATFORM_STATUS    main      SwirldsPlatform: Platform status changed to...
+ * timestamp               log number   log level      marker     thread name     class name         remainder of line
+ */
+public class LogLine implements FormattableString {
+    public final Logger logger = LogManager.getLogger();
+    public static final String CAPTURED_WHITESPACE_REGEX = "(\\s+)";
+    public static final String TIMESTAMP_REGEX = "(\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d+)";
+    public static final String LOG_NUMBER_REGEX = "(\\d+)";
+    public static final String LOG_LEVEL_REGEX = "(TRACE|DEBUG|INFO|WARN|ERROR|FATAL)";
+    public static final String MARKER_REGEX = "([A-Za-z0-9_]+)";
+    public static final String THREAD_NAME_REGEX = "(<<?[^>]+>>?)";
+    public static final String CLASS_THREAD_NAME_REGEX = "([A-Za-z0-9_]+)";
+    public static final String COLON_SPACE_REGEX = "(: )";
+    public static final String REMAINDER_OF_LINE_REGEX = "(.*)";
+
+    public static final TextEffect TIMESTAMP_COLOR = TextEffect.GRAY;
+    public static final TextEffect LOG_NUMBER_COLOR = TextEffect.WHITE;
+    public static final TextEffect LOG_MARKER_COLOR = TextEffect.BRIGHT_BLUE;
+    public static final TextEffect THREAD_NAME_COLOR = TextEffect.BRIGHT_WHITE;
+    public static final TextEffect CLASS_NAME_COLOR = TextEffect.BRIGHT_CYAN;
+    public static final TextEffect HARMLESS_LOG_LEVEL_COLOR = TextEffect.BRIGHT_GREEN;
+    public static final TextEffect WARN_LOG_LEVEL_COLOR = TextEffect.BRIGHT_YELLOW;
+    public static final TextEffect ERROR_LOG_LEVEL_COLOR = TextEffect.BRIGHT_RED;
```

Comment:
Could we make this into a record?

Reply:
You mean a record just containing static constants?